### PR TITLE
Add PostGres support

### PR DESCRIPTION
**Pull request by gareth-ib**
Here ya go :) Postgres doesn't use the same SQL commands to get structure, so I put them in there

### CODE REVIEW

**Comment by gareth-ib** [comment]
I added StyleCI to my fork so it'd match your rules but that seems to cause more issues... do you have a special formatting setup your StyleCI?

**Comment by CristianLlanos** [comment]
Hi, @gareth-ib, thanks for contributing. The style config I'm using for this package is the same as Laravel's.

File: src/Coders/Console/CodeModelsCommand.php [hunk 1]
```diff
@@ -86,7 +95,7 @@ protected function getConnection()
      */
     protected function getSchema($connection)
     {
-        return $this->option('schema') ?: $this->config->get("database.connections.$connection.database");
+        return $this->option('schema');
```
Comment (CristianLlanos) [inline src/Coders/Console/CodeModelsCommand.php]: Why did you change this line?

Comment (gareth-ib) [reply src/Coders/Console/CodeModelsCommand.php]: because there is always something defined in the database config, it'll never be empty. The "mapAll" method that gets called when they don't force a specific schema will be called instead and handle whatever database they are connected to

Comment (CristianLlanos) [reply src/Coders/Console/CodeModelsCommand.php]: Would you mind making different PRs for each functionality you are trying to add? It'll be easier to review. I appreciate you're taking the time to contribute, btw. Thanks.

Comment (CristianLlanos) [reply src/Coders/Console/CodeModelsCommand.php]: I've noticed you are trying to add support for Postgress. That could be one PR. In addition, I see you are trying to refactor some code; that could be another PR. And please try not to change the code style, those changes make it really hard to spot the meaningful changes your are making to this package. I hope you won't mind.

Comment (gareth-ib) [reply src/Coders/Console/CodeModelsCommand.php]: Well the postgres version can't run against the whole active database without that change :) that's why I had to add it

Comment (gareth-ib) [reply src/Coders/Console/CodeModelsCommand.php]: yeah trying to keep it all to just postgres stuff, but PG is so different from MySQL/SQL that I needed more changes. Postgres has a tree separation of Schemas and Databases, as opposed to the other SQL's treating them as the same thing

Comment (gareth-ib) [reply src/Coders/Console/CodeModelsCommand.php]: updated the style ci to use the laravel preset, hopefully that puts back all the style changes it made using the default styleci

Comment (gareth-ib) [reply src/Coders/Console/CodeModelsCommand.php]: cool the style conflicts look resolved. but yeah without this config change to be directly params instead of combining the `database` concept with `schema`, the postgres version can't run. Cuz PG has a setup like `Server.[ Databases ].[ Schemas ].[ tables ]`
as opposed to the mysql style of `Server.[ Databases ].[ tables ]`

Comment (gareth-ib) [reply src/Coders/Console/CodeModelsCommand.php]: so with postgres that means it'd try to run against a schema with the name of the databas

**Comment by gareth-ib** [comment]
might be good to add a `.styleci.yml` file with your `preset: laravel` in it, so that config already exists in the project before anyone runs styleci on it. for cases like what happened to me :)

File: src/Coders/Console/CodeModelsCommand.php (hunk 1 above)
Comment (CristianLlanos) [reply src/Coders/Console/CodeModelsCommand.php]: The community will be very grateful for your contribution :)

Comment (gareth-ib) [reply src/Coders/Console/CodeModelsCommand.php]: glad :) I have other ideas to add you might like... one being adding a type-hint phpdoc replacement for the Eloquent methods, so the IDE can know that things like `$object->where(...)` is going to return that model

**Comment by gareth-ib** [comment]
hey is there anything holding up the PR still?

**Comment by CristianLlanos** [comment]
Hi, Gareth, I need some time to have a thorough look at the PR. I'll try to do it at the weekend. Big thanks for your contribution.

**Comment by gareth-ib** [comment]
oh ok cool, just wanna get it in there so I can tinker with more ideas. maybe i'll just do a branch :)

**Comment by CristianLlanos** [comment]
Hi, Gareth. I'm sorry I wasn't able to merge the PR this week. I'll definitely do it next Sunday.

**Comment by gareth-ib** [comment]
okay cool. I just noticed an issue with tables not getting defined even though I have qualify set to true in the config

```php
    /**
     * @return bool
     */
    public function needsTableName()
    {
        return false === $this->shouldQualifyTableName() ||
            $this->shouldRemoveTablePrefix() ||
            $this->blueprint->table() != Str::plural($this->getRecordName()) ||
            ! $this->shouldPluralizeTableName();
    }
```

that being the reason. I changed that `false` to `true` to fix it. Is there something I'm missing about that statement? It's a pretty complex set of or statements. Is this issue unique to my postgres changes?

**Comment by CristianLlanos** [comment]
I'm so sorry. I went on holidays and I forgot about everything. My bad. I'll try to make some space to review it this week. And thanks again for your contribution. You rock!

**Comment by gareth-ib** [comment]
got a minute? :) doing new projects and hard coding my fork into them is getting problematic

**Comment by CristianLlanos** [comment]
Hi, @gareth-ib, I'm sorry for keeping you waiting this long. @rwdim has made a PR adding support for PostgreSQL (#111). Since it's easier to review I'm about to merge it. However, it'd be amazing if you could have a look at it. Your feedback would be very much appreciated.

**Comment by pursehouse** [comment]
wow. you really wasted my time.

**Comment by gareth-ib** [comment]
so you took so long to merge that someone else also built the same system. thanks for wasting my time.

**Comment by CristianLlanos** [comment]
The reason I hesitate to merge this PR is because there are more changes than what it's suppose to have. Since we don't have tests, there is no guarantee this PR is not breaking anything. Let me suggest contributing each functionality on its own PR. This way its easier to review and spot breaking changes. I'm sorry and grateful to you for taking the time to contribute to this project.

**Comment by gareth-ib** [comment]
because your system doesn't support schemas, so yeah, actual support for postgres means more work

**Comment by CristianLlanos** [comment]
Hi, @gareth-ib, does PostgreSQL support cross-database relationships? Or maybe cross-schema relationships? 🤔

**Comment by gareth-ib** [comment]
@CristianLlanos yes of course PostGres has cross schema relationships.

**Comment by timhaak** [comment]
As they don't seem to be accepting pull requests and maintaining. 

I've forked the repository and merged some of the pull requests.

https://github.com/haakco/reliese-laravel-models

https://packagist.org/packages/haakco/reliese-laravel-models

**Comment by gareth-ib** [comment]
@timhaak hey, thanks for making your branch, I actually have been actively maintaining my own fork now. If you want to PR into that anything you need changed, that would be great
https://github.com/pursehouse/modeler-laravel-eloquent

**Comment by CristianLlanos** [comment]
@gareth-ib, thanks for contributing. PR #111 has been merged adding support for PostgreSQL.
